worlds/yoshisisland/Rules.py

```diff
@@ -0,0 +1,597 @@
+    set_rule(multiworld.get_location("Poochy Ain't Stupid: Level Clear", player), lambda state: state.has('Poochy', player))
+
+    set_rule(multiworld.get_location("Hit That Switch!!: Red Coins", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+    set_rule(multiworld.get_location("Hit That Switch!!: Flowers", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+    set_rule(multiworld.get_location("Hit That Switch!!: Level Clear", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+
+    set_rule(multiworld.get_location("The Impossible? Maze: Red Coins", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph', 'Flashing Eggs'}, player))
+    set_rule(multiworld.get_location("The Impossible? Maze: Flowers", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph'}, player))
+    set_rule(multiworld.get_location("The Impossible? Maze: Stars", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph'}, player))
+    set_rule(multiworld.get_location("The Impossible? Maze: Level Clear", player), lambda state: state.has_all({'Spring Ball', 'Large Spring Ball', 'Mole Tank Morph', 'Helicopter Morph'}, player))
+
+    set_rule(multiworld.get_location("Kamek's Revenge: Red Coins", player), lambda state: state.has_all({'Key', 'Skis', 'Helicopter Morph', '! Switch'}, player) and logic.has_midring(state))
+    set_rule(multiworld.get_location("Kamek's Revenge: Flowers", player), lambda state: state.has_all({'Key', 'Skis', 'Helicopter Morph', '! Switch'}, player) and logic.has_midring(state))
+    set_rule(multiworld.get_location("Kamek's Revenge: Stars", player), lambda state: state.has_all({'! Switch'}, player) or logic.has_midring(state))
+    set_rule(multiworld.get_location("Kamek's Revenge: Level Clear", player), lambda state: state.has_all({'Key', 'Skis', 'Helicopter Morph'}, player))
+
+    set_rule(multiworld.get_location("Castles - Masterpiece Set: Red Coins", player), lambda state: (state.has('Egg Capacity Upgrade', player, 1) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
+    set_rule(multiworld.get_location("Castles - Masterpiece Set: Flowers", player), lambda state: (state.has('Egg Capacity Upgrade', player, 1) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
+    set_rule(multiworld.get_location("Castles - Masterpiece Set: Stars", player), lambda state: logic.has_midring(state) or state.has(('Large Spring Ball'), player))
+    set_rule(multiworld.get_location("Castles - Masterpiece Set: Level Clear", player), lambda state: (state.has('Egg Capacity Upgrade', player, 1) or logic.combat_item(state)) and state.has(('Large Spring Ball'), player))
+
+def set_hard_rules(world, multiworld, player: int, boss_order, luigi_pieces):

```

Comment:
Recommend reworking set_hard_rules to only take world, since multiworld and player you can get from world, and you aren't using boss_order or luigi_pieces